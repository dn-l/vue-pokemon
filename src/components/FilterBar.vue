<template>
  <v-row justify="space-around" align="center" class="pa-2">
    <v-col>
      <v-switch
        :input-value="filterParams.showOnlyFavourite"
        @change="setFilter('showOnlyFavourite', $event)"
        label="Only Favourites"
      />
    </v-col>
    <v-col>
      <v-switch
        :input-value="filterParams.showEvolutions"
        @change="setFilter('showEvolutions', $event)"
        label="Show Evolutions"
      />
    </v-col>
    <v-col>
      <v-select v-model="sortBy" :items="sortByOptions" label="Sort by" />
    </v-col>
  </v-row>
</template>

<script lang="ts">
import Vue from 'vue';
import { SortBy } from '../State';

export default Vue.extend({
  computed: {
    filterParams() {
      return this.$store.state.filterParams;
    },
    sortByOptions() {
      return Object.keys(SortBy);
    },
    sortBy: {
      get() {
        return this.$store.state.sortBy;
      },
      set(sortBy) {
        return this.$store.dispatch('setSortBy', sortBy);
      },
    },
  },

  methods: {
    setFilter(name: string, value: boolean) {
      this.$store.dispatch('setFilter', { [name]: value });
    },
  },
});
</script>
