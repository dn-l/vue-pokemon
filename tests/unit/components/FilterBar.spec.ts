import Vue from 'vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import FilterBar from '@/components/FilterBar.vue';

const localVue = createLocalVue();
Vue.use(Vuetify);
localVue.use(Vuex);
localVue.use(VueRouter);

const store = new Vuex.Store({
  state: { filterParams: {} },
});

describe('FilterBar.vue', () => {
  const wrapper = shallowMount(FilterBar, {
    store,
    localVue,
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
