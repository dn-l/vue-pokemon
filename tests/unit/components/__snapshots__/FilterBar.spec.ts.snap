// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar.vue matches snapshot 1`] = `
<v-row-stub tag="div" align="center" justify="space-around" class="pa-2">
  <v-col-stub tag="div">
    <v-switch-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Only Favourites" ripple="true" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"></v-switch-stub>
  </v-col-stub>
  <v-col-stub tag="div">
    <v-switch-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Show Evolutions" ripple="true" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"></v-switch-stub>
  </v-col-stub>
  <v-col-stub tag="div">
    <v-select-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" appendicon="$vuetify.icons.dropdown" backgroundcolor="" label="Sort by" loaderheight="2" clearicon="$vuetify.icons.clear" type="text" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" nodatatext="$vuetify.noDataText" items="Name,Number,Damage" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]"></v-select-stub>
  </v-col-stub>
</v-row-stub>
`;
